src/storage/v2/vertex_accessor.cpp

```diff
@@ -403,6 +403,10 @@ Result<PropertyValue> VertexAccessor::GetProperty(PropertyId property, View view
   return std::move(value);
 }
 
+Result<uint64_t> VertexAccessor::GetPropertySize(PropertyId property) const {
+  return vertex_->properties.PropertySize(property);
```

Comment:
I think this is not correct. Since we don't have strict schema, the type of property can change and hence its size can change. I think this method needs to accept `View` and get the property value depending on the `View` similar to how `GetProperty` operates. Then you will probably have some duplicated code so please extract that also

Reply:
Yeah not thread safe, need to get shared lock to do anything on properties.
And as Andi said it should be consistent to the isolation level, hence you maybe need to copy + apply deltas.